tools/pyvmi/setup.py

```diff
@@ -1,21 +1,15 @@
-#!/usr/bin/env python
-import commands
-from distutils.core import setup, Extension
+#!/usr/bin/env python3
 
-pkgconfig_cflags = commands.getoutput ("pkg-config --cflags glib-2.0 libvmi")
-pkgconfig_include_flags = commands.getoutput ("pkg-config --cflags-only-I glib-2.0 libvmi")
-pkgconfig_include_dirs = [ t[2:] for t in pkgconfig_include_flags.split() ]
-pkgconfig_lflags = commands.getoutput ("pkg-config --libs-only-l glib-2.0 libvmi")
-pkgconfig_libs = [ t[2:] for t in pkgconfig_lflags.split() ]
-pkgconfig_biglflags = commands.getoutput ("pkg-config --libs-only-L glib-2.0 libvmi")
-pkgconfig_ldirs = [ t[2:] for t in pkgconfig_biglflags.split() ]
 
-pyvmimod = Extension('pyvmi', sources=['pyvmi.c'],
-                    include_dirs = pkgconfig_include_dirs,
-                    library_dirs = pkgconfig_ldirs,
-                    libraries = pkgconfig_libs,
-                    extra_compile_args=[pkgconfig_cflags])
+from setuptools import setup
 
-setup(name='PyVmi', version='1.1',
-      description = 'Python interface to LibVMI',
-      ext_modules = [pyvmimod])
+
+setup(
```

Comment:
Your package won't include the external files meaning it won't compile on another host. You need to add them via the `data_files` keyarg.

You also probably need to include the `ext_modules` or `ext_package` keywords. Check the [`distutils`](https://docs.python.org/3.6/distutils/setupscript.html#describing-extension-modules) for more information.

I'd recommend you to try to run
```
$ python3 setup.py sdist
``` 
Then grab the resulting `tar.gz` and try to install on a separate host to test it builds correctly.

You can check a complete example [here](https://github.com/noxdafox/clipspy/blob/master/setup.py#L66).

Reply:
good catch !
i had to create a `libvmi/` package, and use `package_data` to include the `libvmi_cdef.h`